pkg/client/client.go

```diff
@@ -261,3 +271,18 @@ func MergeAnnotations(toolchainObject client.Object, newAnnotations map[string]s
 	}
 	toolchainObject.SetAnnotations(annotations)
 }
+
+// ApplyUnstructuredObjects applies the given Unstructured objects on the cluster.
+func ApplyUnstructuredObjects(ctx context.Context, cl client.Client, unstructuredObjects []*unstructured.Unstructured, newLabels map[string]string) error {
```

Comment:
I'd suggest to rename this function so that it's more explicit what `newLabels` is used for. Maybe something like `ApplyUnstructuredObjectsWithNewLabels`. WDYT?

Reply:
Makes sense. Renamed in https://github.com/codeready-toolchain/toolchain-common/pull/373/commits/59712631cdb878aaaceb7eeb1474dce8fc12e191